twitchio/ext/sounds/queuemanager.py

```diff
@@ -0,0 +1,155 @@
+import asyncio
+from twitchio.ext import sounds
+from typing import Optional
+
+
+class AudioQueueManager:
+    """
+    Manages a queue of audio files to be played sequentially with optional repeat and pause functionalities.
+
+    Attributes
+    ----------
+    queue: asyncio.Queue[:class:`str`]
+        A queue to hold paths of audio files to be played.
+    is_playing: :class:`bool`
+        Indicates whether an audio file is currently being played.
+    repeat_queue: :class:`bool`
+        If True, adds the current playing audio file back to the queue after playing.
+    queue_paused: :class:`bool`
+        If True, pauses the processing of the queue.
+    player: :class:`sounds.AudioPlayer`
+        An instance of AudioPlayer to play audio files.
+    current_sound: :class:`str`
+        Path of the currently playing audio file.
+    """
+
+    def __init__(self, repeat_queue: Optional[bool] = True) -> None:
+        """
+        Initializes an instance of AudioQueueManager with an empty queue and default settings.
+
+        Parameters
+        ----------
+        repeat_queue: Optional[:class:`bool`]
+            If True, adds the current playing audio file back to the queue after playing, by default True
+        """
+        self.queue: asyncio.Queue[str] = asyncio.Queue()
+        self.is_playing: bool = False
+        self.repeat_queue: bool = repeat_queue if repeat_queue is not None else True
```

Comment:
When can repeat_queue be None? Is this not just a True/False scenario?
The only other reference I see to it is checking whether it is True or not, not whether it is None.

Reply:
Good point, that was my mistake